src/coreclr/jit/assertionprop.cpp

```diff
@@ -2657,21 +2657,18 @@ GenTree* Compiler::optVNBasedFoldExpr_Call(BasicBlock* block, GenTree* parent, G
         case CORINFO_HELP_ISINSTANCEOFANY:
         case CORINFO_HELP_ISINSTANCEOFINTERFACE:
         {
-            GenTree* castClsArg = call->gtArgs.GetUserArgByIndex(0)->GetNode();
-            GenTree* castObjArg = call->gtArgs.GetUserArgByIndex(1)->GetNode();
-
-            if ((castObjArg->gtFlags & GTF_ALL_EFFECT) != 0)
-            {
-                // It won't be trivial to properly extract side-effects from the call node.
-                // Ideally, we only need side effects from the castClsArg argument as the call itself
-                // won't throw any exceptions. But we should not forget about the EarlyNode (setup args)
-                return nullptr;
-            }
```

Comment:
Why would `gtExtractSideEffList(call, ..., /* ignoreRoot */ true)` not be sufficient here? I don't think extracting the side effects of the operands of a call is all that hard.

Reply:
@jakobbotsch I am not sure I understand how it can work, say I have
```
call(arg0, arg1, arg2)
```
I want to fold it to just `arg1` (or to e.g. `arg1*arg1` in case of `Math.Pow(x, 2)`) and say arg1 has a side-effect in it.
if we just return arg1 node and wrap with `gtExtractSideEffList` we'll just duplicate the side effect

Reply:
Ah okay, I see -- that makes sense.